oio/xcute/common/job.py

```diff
@@ -30,23 +30,63 @@ def process(self, task_id, task_payload):
 class XcuteJob(object):
 
     JOB_TYPE = None
+    TASK_CLASS = None
+
+    DEFAULT_TASKS_PER_SECOND = 32
+    MAX_TASKS_BATCH_SIZE = 512
 
-    @staticmethod
-    def sanitize_params(params):
+    def __init__(self, conf, logger=None):
+        self.conf = conf
+        self.logger = logger or get_logger(self.conf)
+
+    def sanitize_config(self, job_config):
+        """
+            Validate and sanitize the job configuration
+            Ex: cast a string as integer, set a default
+            Also return the lock id if there is one
+        """
+        sanitized_job_config = dict()
+
+        tasks_per_second = int_value(
+            job_config.get('tasks_per_second'),
+            self.DEFAULT_TASKS_PER_SECOND)
+        sanitized_job_config['tasks_per_second'] = tasks_per_second
+
+        tasks_batch_size = int_value(
+            job_config.get('tasks_batch_size'), None)
+        if tasks_batch_size is None:
+            if tasks_per_second > 0:
+                tasks_batch_size = min(
+                    tasks_per_second, self.MAX_TASKS_BATCH_SIZE)
+            else:
+                tasks_batch_size = self.MAX_TASKS_BATCH_SIZE
+        elif tasks_batch_size < 1:
+            raise ValueError('Tasks batch size should be positive')
+        elif tasks_batch_size > self.MAX_TASKS_BATCH_SIZE:
+            raise ValueError('Tasks batch size should be less than %d' %
+                             self.MAX_TASKS_BATCH_SIZE)
+        sanitized_job_config['tasks_batch_size'] = tasks_batch_size
+
+        sanitized_job_params, lock = self.sanitize_params(
+            job_config.get('params') or dict())
```

Comment:
`.get('params', dict())`

Reply:
If the `params` key is in `job_config`, but its value is `None`, we must be sure to have a `dict`.